### PR TITLE
chore(flakiness): Flakiness Dashboard fixes

### PR DESCRIPTION
**Pull request by aslushnikov**
- fix `FLAKINESS_DASHBOARD_BUILD_URL` to point to a task instead of a build
- do not pretty-print `dashboard.json` when serializing flakiness results
- filter out 'COVERAGE' test(s) so that they don't add up to `dashboard.json` payload. These are useless
- validate certain important options of flakiness dashboard
- more logging to STDOUT to actually say which repo and what branch is getting used
- enhance commit message with a build URL
- use a more compact format for JSON. For 100 runs of 700 tests it yields 21MB json instead of 23MB.
- bump default builds number to 100

### CODE REVIEW

**Review by MedRedha — APPROVED** [review]

**Review by MedRedha — APPROVED** [review]

File: utils/flakiness-dashboard/FlakinessDashboard.js [hunk 1]
```diff
@@ -84,19 +94,31 @@ class Dashboard {
...
+      url: test.u,
+      result: test.r,
+    }))));
     return new Dashboard(name, dashboardPath, builds, options);
   }
 
   async saveJSON() {
     const data = { version: DASHBOARD_VERSION };
     data.builds = this._builds.map(build => ({
-      timestamp: build._timestamp,
```
Comment (JoelEinbinder) [inline utils/flakiness-dashboard/FlakinessDashboard.js:94]: Uh, git compresses things. I think this is unnecessary.

Comment (aslushnikov) [reply utils/flakiness-dashboard/FlakinessDashboard.js:94]: that's for the custom flakienss viewer. This saves ~100kb in my tests.

Comment (JoelEinbinder) [reply utils/flakiness-dashboard/FlakinessDashboard.js:94]: How are you operating where an uncompressed 100kb matters? I'd much rather have a readable data format.

Comment (aslushnikov) [reply utils/flakiness-dashboard/FlakinessDashboard.js:94]: > uncompressed 100kb matters

Compressed 100kb

**Review by JoelEinbinder — APPROVED** [review]

**Pull request merged**